src/FFmpegReader.cpp

```diff
@@ -978,7 +986,16 @@ std::shared_ptr<Frame> FFmpegReader::ReadStream(int64_t requested_frame) {
 			// Process Audio Packet
 			ProcessAudioPacket(requested_frame);
 		}
-		
+
+		// Remove unused packets (sometimes we purposely ignore video or audio packets,
+		// if the has_video or has_audio properties are manually overridden)
+		if ((!info.has_video && packet && packet->stream_index == videoStream) ||
+			(!info.has_audio && packet && packet->stream_index == audioStream)) {
+			RemoveAVPacket(packet);
+			packet = NULL;
+			packets_decoded++;
```

Comment:
This makes a lot of sense that we're discarding packets from streams we want to ignore.